### PR TITLE
LMS - Who lives here section

### PR DESCRIPTION
**Pull request by jonnyshaw89**
Have updated the Who lives here section to meet the slides

Also fixes navigation to work with the multiple repeating groups

### CODE REVIEW

File: data/en/lms_2.json [hunk 1]
```diff
@@ -33,94 +33,103 @@
...
-                                            "description": "The address above is my main residence"
+                                            "label": "Yes, the above address is my main residence",
+                                            "value": "Yes"
                                         },
                                         {
-                                            "label": "No",
+                                            "label": "No, the above address is not my main residence",
                                             "value": "No",
-                                            "description": "The address above is not my main residence.  For example it is my old address, second home, holiday home or business address"
+                                            "description": "For examples it is my old address, second home, holiday home, business address or is not familiar to me"
```
Comment (bitdivision) [inline data/en/lms_2.json:101]: For example

Comment (bitdivision) [reply data/en/lms_2.json:101]: I've commented on the spec, Donna will need to change it.

File: data/en/lms_2.json [hunk 2]
```diff
@@ -3412,196 +3575,147 @@
...
+                                            "id": "working-days-answer",
+                                            "condition": "contains",
+                                            "value": "Did not work this week"
+                                        }
+                                    ]
                                 }
                             },
                             {
                                 "goto": {
-                                    "block": "unpaid-overtime-w10",
```
Comment (bitdivision) [inline data/en/lms_2.json:3477]: This is just a bad diff I think?

**Review by bitdivision — COMMENTED** [review]
Obviously address piping and navigation aren't in yet. 

I've done a little bit of testing on the routing and flow and it all seems to be as expected.

**Review by MebinAbraham — COMMENTED** [review]
Person icon needs to be added instead of bullets for listing peoples names?

File: data/en/lms_2.json [hunk 3]
```diff
@@ -53,93 +53,103 @@
...
                         },
                         {
                             "type": "Question",
                             "id": "address-check-block",
-                            "description": "",
+                            "description": "{{metadata['ru_name']}}",
                             "questions": [{
                                 "id": "address-check-question",
-                                "title": "Is this address your main residence?",
+                                "title": "Is this address your main residence",
```
Comment (MebinAbraham) [inline data/en/lms_2.json]: Does it not need the `?`

File: data/en/lms_2.json [hunk 4]
```diff
@@ -53,93 +53,103 @@
...
+                                        "type": "TextField"
+                                    }
                                 ]
                             }]
                         },
                         {
                             "type": "Question",
                             "id": "address-check-block",
-                            "description": "",
+                            "description": "{{metadata['ru_name']}}",
```
Comment (MebinAbraham) [inline data/en/lms_2.json]: Bad merge?. Not meant to be using `ru_name` anymore?

**Review by MebinAbraham — COMMENTED** [review]
Question:
Do we require the driver questions to be mandatory?:
`/other-household-member-group/0/other-household-member-anyone-else-block`
and
`/student-household-member-group/0/student-household-member-anyone-else-block`
are both mandatory in schema but not in the spec. Is this a requirement for `repeat_until` cause we don't treat no answer the same way?

**Review by MebinAbraham — CHANGES_REQUESTED** [review]
Questions for slide 10 and 11 are mandatory in the schema but not according to spec.

File: data/en/lms_2.json [hunk 5]
```diff
@@ -556,15 +716,15 @@
                         "questions": [{
                             "id": "time-away-question",
                             "titles": [{
-                                    "value": "Is <em>{{[answers['first-name'][group_instance], answers['last-name'][group_instance]] | format_household_name }}</em> currently away from {{ metadata['display_address'] }} for a continuous period of 6 months or more?",
+                                    "value": "Is <em>{{[group_instances[group_instance_id]['primary-household-member-first-name'], group_instances[group_instance_id]['primary-household-member-last-name'], group_instances[group_instance_id]['other-household-member-first-name'], group_instances[group_instance_id]['other-household-member-last-name'], group_instances[group_instance_id]['student-household-member-first-name'], group_instances[group_instance_id]['student-household-member-last-name']] | format_household_name}}</em> currently away from {{ metadata['ru_name'] }} for a continuous period of 6 months or more?",
```
Comment (dcdarrell9) [inline data/en/lms_2.json]: Same here with `ru_name`

File: app/questionnaire/navigation.py [hunk 6]
```diff
@@ -146,18 +145,17 @@ def _generate_item(self, name, completed, location, highlight, repeating=False):
             'repeating': repeating,
         }
 
-    def _generate_link_names(self, label_answer_ids):
-        link_names = defaultdict(list)
+    def _generate_link_name(self, label_answer_ids, location):
+        link_names = []
+
+        group_instance_id = get_group_instance_id(self.schema, self.answer_store, location)
```
Comment (bitdivision) [inline app/questionnaire/navigation.py:151]: 👍

**Comment by jonnyshaw89** [comment]
@MebinAbraham can you re-review?

**Comment by dcdarrell9** [comment]
Sooo, when you navigate back to `About Household` If you change the name then it works fine, but then if you try to 'add' any more members to the household, it will overwrite the next in the list. 

For example: 
- If you had added -> 'darrell cox' / 'jon shaw' / 'mebin abraham'
- Return to About Household and be asked about 'darrell cox' name -> change to 'Darrell Cox' -> save
- On the next screen it will ask if you want to add another member -> add another member 'Andrew MadDog' -> save -> 'jon shaw' will have been overwritten

Not sure what the expected functionality is, or what we should be doing with this

**Review by bitdivision — APPROVED** [review]

**Review by MebinAbraham — COMMENTED** [review]
Some routing on slide 5 (Hholdchk) seems to be missing.
```If no and answer to ineligbletype = ‘student accommodation rented from a private landlord’ then go to sharecooking```

Routing goes to `Addppl` instead of `sharecooking`

So:
- Person 1 `(/primary-household-member-group/0/primary-household-member-block)`
- No, the above address is not my main residence `(/primary-household-member-group/0/address-check-block)`
- Student accommodation rented through a private landlord `(/address-type-group/0/address-type-block)`
- No, this is not the main residence of any other adults, children or babies `(/other-household-member-group/0/other-household-member-anyone-else-block)`

Goes to: `/student-household-member-group/0/student-household-member-anyone-else-block`
but should go to: `/share-group/0/share-cooking-block`?

**Comment by jonnyshaw89** [comment]
@MebinAbraham can you check again?

**Comment by MebinAbraham** [comment]
@jonnyshaw89 Not sure if it was the recent change that @boxadesign or because of the groups/block structure, but some blocks don't have `Previous` links such as the sharecooking.

Also, the mandatoriness doesn't match the spec. Apart from that, the flow seems to be as expected.

**Comment by jonnyshaw89** [comment]
@MebinAbraham any block at the beginning of a group doesnt gave a Previous link.
can you be more specific about missing mandatory?

**Comment by MebinAbraham** [comment]
@jonnyshaw89 sharecooking

**Review by MebinAbraham — APPROVED** [review]

**Pull request merged**